bindings/alicloud/dingtalk/webhook/webhook.go

```diff
@@ -0,0 +1,218 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	webhookContentType       = "application/json"
+	defaultHTTPClientTimeout = time.Second * 30
+)
+
+type DingTalkWebhook struct {
+	logger     logger.Logger
+	settings   Settings
+	httpClient *http.Client
+}
+
+type webhookResult struct {
+	ErrCode int    `json:"errcode"`
+	ErrMsg  string `json:"errmsg"`
+}
+
+type outgoingWebhook struct {
+	handler func(*bindings.ReadResponse) ([]byte, error)
+}
+
+var webhooks sync.Map //nolint:gochecknoglobals
```

Comment:
Was there any reason why you chose `sync.Map` here instead of a built-in map and Mutex/RWMutex?

https://medium.com/@deckarep/the-new-kid-in-town-gos-sync-map-de24a6bf7c2c